File: app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/adapter/PaslonAdapter.java
package com.berkatfaatulohalawa1711010164.facevoting.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.berkatfaatulohalawa1711010164.facevoting.R;
import com.berkatfaatulohalawa1711010164.facevoting.config.Constants;
import com.berkatfaatulohalawa1711010164.facevoting.model.PaslonModel;
import com.berkatfaatulohalawa1711010164.facevoting.ui.Detailpaslon;
import com.bumptech.glide.Glide;
import com.bumptech.glide.request.RequestOptions;

import java.util.List;

public class PaslonAdapter extends RecyclerView.Adapter<PaslonAdapter.MyViewHolder>{
    List<PaslonModel> mPaslonList;
    private final Context context;

    public PaslonAdapter(Context context,List<PaslonModel> mPaslonList) {
        this.mPaslonList = mPaslonList;
        this.context = context;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View mView = LayoutInflater.from(parent.getContext()).inflate(R.layout.single_row_paslon, parent, false);
        PaslonAdapter.MyViewHolder holder;
        holder = new MyViewHolder(mView);
        return holder;
    }

    @Override
    public void onBindViewHolder (@NonNull MyViewHolder holder,final int position){
        Glide.with(context)
                .load(Constants.IMAGES_URL+mPaslonList.get(position).getPhoto1_paslon())
                .apply(new RequestOptions().error(R.drawable.profil))
                .into(MyViewHolder.mGambarKetua);
        Glide.with(context)
                .load(Constants.IMAGES_URL+mPaslonList.get(position).getPhoto2_paslon())
                .apply(new RequestOptions().error(R.drawable.profil))
                .into(MyViewHolder.mGambarWakil);
        holder.mTxtKetua.setText(mPaslonList.get(position).getKetua_paslon());
        holder.mTxtWakil.setText(mPaslonList.get(position).getWakil_paslon());
        holder.mTxtJudul.setText(mPaslonList.get(position).getJudul_paslon());
        holder.itemView.setOnClickListener(v -> {
            Intent mIntent = new Intent(v.getContext(), Detailpaslon.class);
            mIntent.putExtra("idPaslon",mPaslonList.get(position).getId_paslon());
            mIntent.putExtra("idKategori",mPaslonList.get(position).getId_kategori());
            mIntent.putExtra("judulPaslon",mPaslonList.get(position).getJudul_paslon());
            mIntent.putExtra("ketua",mPaslonList.get(position).getKetua_paslon());
            mIntent.putExtra("wakil",mPaslonList.get(position).getWakil_paslon());
            mIntent.putExtra("photo_ketua",mPaslonList.get(position).getPhoto1_paslon());
            mIntent.putExtra("photo_wakil",mPaslonList.get(position).getPhoto2_paslon());
            mIntent.putExtra("visi_misi",mPaslonList.get(position).getVisi_misi());
            mIntent.putExtra("profil_ketua",mPaslonList.get(position).getProfil_catum());
            mIntent.putExtra("profil_wakil",mPaslonList.get(position).getProfil_cawatum());
            mIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
            v.getContext().startActivity(mIntent);
        });
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {
        @SuppressLint("StaticFieldLeak")
        private static ImageView mGambarKetua,mGambarWakil;
        private final TextView mTxtKetua, mTxtWakil,mTxtJudul;
        MyViewHolder(@NonNull View itemView) {
            super(itemView);
            mTxtKetua = itemView.findViewById(R.id.txt_ketua);
            mTxtWakil = itemView.findViewById(R.id.txt_wakil);
            mTxtJudul = itemView.findViewById(R.id.txt_Judul);
            mGambarKetua = itemView.findViewById(R.id.gbr_ketua);
            mGambarWakil  = itemView.findViewById(R.id.gbr_wakil);
        }
    }

    @Override
    public int getItemCount () {
        return mPaslonList.size();
    }

    public void replaceData(List<PaslonModel> paslonModels) {
        this.mPaslonList = paslonModels;
        notifyDataSetChanged();
    }
}

File: android/facevoting/app/src/main/java/com/berkatfaatulohalawa1711010164/facevoting/adapter/PaslonAdapter.kt
package com.berkatfaatulohalawa1711010164.facevoting.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.berkatfaatulohalawa1711010164.facevoting.R
import com.berkatfaatulohalawa1711010164.facevoting.config.Constants
import com.berkatfaatulohalawa1711010164.facevoting.model.PaslonModel
import com.berkatfaatulohalawa1711010164.facevoting.ui.Detailpaslon
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions

class PaslonAdapter(
    private val context: Context,
    private var mPaslonList: List<PaslonModel>
) : RecyclerView.Adapter<PaslonAdapter.MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val mView = LayoutInflater.from(parent.context).inflate(R.layout.single_row_paslon, parent, false)
        return MyViewHolder(mView)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        val item = mPaslonList[position]
        Glide.with(context)
            .load(Constants.IMAGES_URL + item.photo1_paslon)
            .apply(RequestOptions().error(R.drawable.profil))
            .into(MyViewHolder.mGambarKetua)
        Glide.with(context)
            .load(Constants.IMAGES_URL + item.photo2_paslon)
            .apply(RequestOptions().error(R.drawable.profil))
            .into(MyViewHolder.mGambarWakil)
        holder.mTxtKetua.text = item.ketua_paslon
        holder.mTxtWakil.text = item.wakil_paslon
        holder.mTxtJudul.text = item.judul_paslon
        holder.itemView.setOnClickListener { v ->
            val mIntent = Intent(v.context, Detailpaslon::class.java).apply {
                putExtra("idPaslon", item.id_paslon)
                putExtra("idKategori", item.id_kategori)
                putExtra("judulPaslon", item.judul_paslon)
                putExtra("ketua", item.ketua_paslon)
                putExtra("wakil", item.wakil_paslon)
                putExtra("photo_ketua", item.photo1_paslon)
                putExtra("photo_wakil", item.photo2_paslon)
                putExtra("visi_misi", item.visi_misi)
                putExtra("profil_ketua", item.profil_catum)
                putExtra("profil_wakil", item.profil_cawatum)
                addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP)
            }
            v.context.startActivity(mIntent)
        }
    }

    override fun getItemCount(): Int = mPaslonList.size

    fun replaceData(paslonModels: List<PaslonModel>) {
        mPaslonList = paslonModels
        notifyDataSetChanged()
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val mTxtKetua: TextView = itemView.findViewById(R.id.txt_ketua)
        val mTxtWakil: TextView = itemView.findViewById(R.id.txt_wakil)
        val mTxtJudul: TextView = itemView.findViewById(R.id.txt_Judul)

        companion object {
            @SuppressLint("StaticFieldLeak")
            lateinit var mGambarKetua: ImageView
            @SuppressLint("StaticFieldLeak")
            lateinit var mGambarWakil: ImageView
        }

        init {
            mGambarKetua = itemView.findViewById(R.id.gbr_ketua)
            mGambarWakil = itemView.findViewById(R.id.gbr_wakil)
        }
    }
}
